packages/app/features/Widget/components/AptosTokensTab/DirectTransfer.tsx

```diff
@@ -18,21 +19,40 @@ const indicatorSize = 16;
 const DirectTransfer: FC<Props> = ({ pubkey, directTransfer, fee }) => {
 	const [disabled, setDisabled] = useState(false);
 
+	const handleOnPasscodeComplete = async (passcode: string) => {
+		const res = {} as ResponsePayload;
+
+		let privateKey;
+		try {
+			privateKey = await utils.getPrivateKey(Networks.aptos, passcode);
+		} catch {
+			res.responseCode = ResponseCode.WRONG_PASSCODE;
+			return res;
+		}
+
+		const transaction = JSON.stringify({
+			pubkey,
+			directTransfer: !directTransfer,
+		} satisfies aptosHandlers.AptosDirectTransferPayload);
+
+		try {
+			res.signatureString = aptosHandlers.handleUpdateDirectTransfer(
+				privateKey,
+				transaction,
+			);
+			res.responseCode = ResponseCode.SUCCESS;
+		} catch (error) {
+			res.responseCode = ResponseCode.ERROR;
+		}
+
+		return res;
```

Comment:
this type of function can not be put in here, it must be implemented in `app`. In mobile, we dont have messaging module, so `createAndSend` in mobile will take action to `private key` and make a response. On the other side, web, these handlings occur in the kernel, `createAndSend` in web just passing payload with construct transaction and send to kernel via messaging. So this method need to be reimplemented to make it universal